### PR TITLE
Add api to load a vega-lite spec into voyager.

### LINKED ISSUE #377: API for loading visual encoding
For example, if the electron app wanna support loading .vl.json file, we need to be able to update the shelf

### PR DESCRIPTION
**Pull request by tafsiri**
Add api to load a vega-lite spec into voyager. Note that this will clear any previously set data. Closes #377

Ping @kanitw. Finally got this going, tests took longer to debug/make run correctly than I expected.

### CODE REVIEW

File: src/components/app.tsx [hunk 1]
```diff
@@ -89,6 +103,60 @@ class AppBase extends React.PureComponent<Props, {}> {
...
+      allErrors: true,
+      extendRefs: 'fail'
+    });
+    ajv.addMetaSchema(draft4Schemas, 'http://json-schema.org/draft-04/schema#');
+
+    const validateVl = ajv.compile(vlSchema);
+    const valid = validateVl(spec);
+
+    if (!valid) {
+      throw new Error("Invalid Schema:" + validateVl.errors.toString());
```
Comment (kanitw) [inline src/components/app.tsx]: Invalid spec?  (Schema is not invalid, but the spec does not conform to the schema?)

File: src/components/app.tsx [hunk 2]
```diff
@@ -89,6 +103,60 @@ class AppBase extends React.PureComponent<Props, {}> {
...
+
+    const validSpec: TopLevel<FacetedCompositeUnitSpec> = spec as TopLevel<FacetedCompositeUnitSpec>;
+
+
+    if (!isUnitSpec(validSpec)) {
+      throw new Error("Voyager does not support layered or multi-view vega-lite specs");
+    }
+
+    if (validSpec.data) {
+      const setDataAction = this.setData(validSpec.data) as Promise<any>;
```
Comment (kanitw) [inline src/components/app.tsx]: not sure why do you have to cast `as Promise<any>` here.   

I wonder if we should just void casting by just doing
```
this.setData(validSpec.data)
  .then(...
```

File: src/components/app.tsx (hunk 1 above)
Comment (tafsiri) [inline src/components/app.tsx]: typo. got it.

File: src/components/app.tsx (hunk 2 above)
Comment (tafsiri) [inline src/components/app.tsx]: good catch

**Comment by tafsiri** [comment]
@kanitw made changes according to your suggestions.

**Comment by kanitw** [comment]
Thanks!

**Pull request merged**